### PR TITLE
optimized commit ts: remove the cost of getting commitTS

### PR DESCRIPTION
**Pull request by alivxxx**
The commit ts now is max(start_ts, key_last_read_ts[k1], key_last_read_ts[k2]....) + 1.
PTAL @coocood @siddontang @foreverbell @GregoryIan

### CODE REVIEW

File: OptimizedCommitTS/OptimizedCommitTS.tla [hunk 1]
```diff
@@ -45,15 +45,15 @@ VARIABLES key_lock
...
 VARIABLES key_write
 
-\* Two auxiliary variables for verifying snapshot isolation invariant. These
-\* variables should not appear in a real-world implementation.
-\*
 \* $key_last_read_ts[k]$ denotes the last read timestamp for key $k$, this
-\* should be monotonic.
-\*
+\* should be monotonic. It can be used to verfiy snapshot isolation invariant
+\* and cacluate commit ts. In a real-world implementation, each region will have
```
Comment (foreverbell) [inline OptimizedCommitTS/OptimizedCommitTS.tla]: typo: calculate

File: OptimizedCommitTS/OptimizedCommitTS.tla [hunk 2]
```diff
@@ -251,10 +254,11 @@ Prewrite(c) ==
   /\ client_state[c] = "prewriting"
   /\ IF client_key[c].pending = {}
      THEN  \* all keys have been pre-written
-       /\ next_ts' = next_ts + 1
        /\ client_state' = [client_state EXCEPT ![c] = "committing"]
-       /\ client_ts' = [client_ts EXCEPT ![c].commit_ts = next_ts']
-       /\ UNCHANGED <<key_vars, client_key>>
+       \* The commit_ts is set to start_ts at first, then updated at every prewrite,
```
Comment (foreverbell) [inline OptimizedCommitTS/OptimizedCommitTS.tla]: I think this comment should be near the definition of client_ts.

**Review by foreverbell — COMMENTED** [review]
Rest LGTM

**Comment by siddontang** [comment]
do we want to use this in TiDB?

**Comment by alivxxx** [comment]
@siddontang Yes.

**Comment by siddontang** [comment]
Seem that Now TiKV will hold the latest read commit now and return to TiDB?

**Comment by alivxxx** [comment]
@siddontang Yes.

**Comment by siddontang** [comment]
seem that now we may have the same start timestamp, is it ok?

**Comment by alivxxx** [comment]
The start timestamp is unique.

**Comment by siddontang** [comment]
@zhangjinpeng1987 Do we do any assumption that the commit ts must be unique?

**Comment by zhangjinpeng87** [comment]
>@zhangjinpeng1987 Do we do any assumption that the commit ts must be unique?

No, we use start_ts to indicate a transaction.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**